### PR TITLE
esp-idf: Make it possible to disable the RGB565 color swap via Kconfig

### PR DESCRIPTION
**Pull request by tronical**
Add an option to disable the swap to big-endian at compile-time.

### CODE REVIEW

**Comment by tronical** [comment]
This is a much less intrusive alternative to #5344 and mirrors the LVGL approach (which also uses a compile-time Kconfig option).

**Comment by tronical** [comment]
I could implement the byte-swap also for double-buffering, but I see two problems:

2) Double-buffering makes only real sense when there are two buffers allocated that the display controller can read out of directly (so esp_lcd_panel_draw_bitmap is called for the entire buffer and the implementation basically ends up swapping out display pointers, like the esp_ld dsi panel driver, for example). In such a configuration, it would be even weirder to have the CPU configured little-endian and the display controller that reads from the same RAM expect big-endian. It doesn't make sense to me, and I don't see any existing example in esp-idf do that either. This was btw also the reason why the esp32-p4 board showed wrong colors: cpu is little endian, display controller reads from PSRAM expecting little endian, we were however doing the wrong byte swap. So I think double buffering and byte swap doesn't make sense.

2) Say we decided that it still makes sense, then we can't introduce that option without breaking compatibility :-(. Whether or not to swap is not something we can detect, the user has to decide. The existing code assumes that slint-esp.cpp is used with a display that expects big endian, so we default to true in this PR (as well as the other). If we implemented this for double buffered setups, then the default would be wrong (would break existing code bases, because they must unselect the byte swap in their config).

The cleanest would be to have byte swapping disabled by default. Then we could implement it also for double buffering (even if it doesn't make sense ;), but if we disable byte swap by default we break existing configs because they need to add SLINT_COLOR_SWAP_16 to their sdkconfig. (the same of course if we use a C++ API instead)

To summarise:

We could either break compatibility with existing setups that rely on the byte swapping and introduce an option that's off by default but allows opting into it. Then we could also implement byte swap for double buffering.

Or we could take this patch, not break anything, but have a slightly less sensible default (on paper) and no support for double buffering.

I wish we could do the first, but I'm not sure how.

**Comment by ogoffart** [comment]
That's where the "runtime" setting is preferable because we can make the old overloads do the right thing.

But yes, I agree that the swap mostly need to be done only for SPI devices where double buffering makes no sense. (Ideally the driver should do the swapping instead of us.

**Review by ogoffart — APPROVED** [review]

**Comment by tronical** [comment]
Yep, and given that some drivers support this, it feels more like a compile time decision. But good point about the overloads, that permits us to do the right thing and maintain compatibility - at the expense of it being an if(). But that's fine :). Alright, I'll review the other PR next week. Thanks for brainstorming with me :)
